src/rgw/rgw_rados.h

```diff
@@ -1313,10 +1313,12 @@ class RGWRados
   RGWDataChangesLog *data_log;
 
   virtual ~RGWRados() {
-    for (uint32_t i=0; i < num_rados_handles; i++) {
-      if (rados[i]) {
-        rados[i]->shutdown();
-        delete rados[i];
+    if (rados){
```

Comment:
Instead of a NULL check on 'rados', isn't it better marking each rados instance in 'init_rados()' to NULL, such that we don't need to touch this file at all. In other words, 'rados[i] = NULL' in init_rados().